hedera-node/hedera-app/src/xtest/java/token/RoyaltyCollectorAutoAssociationXTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package token;
+
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.base.TokenID;
+import com.hedera.hapi.node.state.common.EntityIDPair;
+import com.hedera.hapi.node.state.token.Account;
+import com.hedera.hapi.node.state.token.Token;
+import com.hedera.hapi.node.state.token.TokenRelation;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import java.util.Map;
+
+public class RoyaltyCollectorAutoAssociationXTest extends AbstractTokenXTest {
```

Comment:
Per our discussion offline please add an override for `assertExpectedAccounts` and/or `assertExpectedTokens` and/or `assertExpectedNfts` that simply have a comment that says "this is where validation of assesed royaltys should occur".

IOW an xtest should go deeper than just transaction response code to actually check the state, fees, etc.  (Same for the `PausesXTest` for example: it should really be checking that the state of the nfts reflects the pause.)  But a) this sample doesn't need to provide it yet and b) we should develop (as we discussed) helper methods that make these validations smooth and easy.)

Reply:
Added overrides with such comments.